src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -360,25 +386,35 @@ def step(carry):
                     A=A,
                     b=b,
                 )
+
                 params = optax.apply_updates(params, updates)
+
                 return params, state
 
-            def continuing_criterion(carry):
+            def continuing_criterion(carry: OptimizationCarry) -> jtp.Bool:
+
                 _, state = carry
+
                 iter_num = optax.tree_utils.tree_get(state, "count")
                 grad = optax.tree_utils.tree_get(state, "grad")
                 err = optax.tree_utils.tree_l2_norm(grad)
+
                 return (iter_num == 0) | ((iter_num < maxiter) & (err >= tol))
 
-            init_carry = (init_params, opt.init(init_params))
             final_params, final_state = jax.lax.while_loop(
                 continuing_criterion, step, init_carry
             )
+
             return final_params, final_state
 
+        # ======================================
+        # Compute the contact forces with L-BFGS
+        # ======================================
+
+        # Initialize the optimized forces with a linear Hunt/Crossley model.
```

Comment:
Considering the no-side-effects JAX pattern, I'm a bit hesitant on this type of warm start since it would introduce an extra state. While using the last solution is the optimal warm start, before introducing the complexity of this extra state, we should understand if using a simpler method like the existing one provides a comparable speed-up.